File: stage1/s1_var_05.js
// S1_VAR_05 — Array vs object
function isArray(value) {
  return Array.isArray(value);
}

// Tests
console.log(isArray([])); // true
console.log(isArray([1, 2, 3])); // true
console.log(isArray({})); // false
console.log(isArray({ a: 1 })); // false
console.log(isArray(null)); // false

File: stage1/A/S1_VAR_05.py
# S1_VAR_05.py

def is_truthy(v):
    return bool(v)


def main():
    tests = [
        0,
        1,
        "",
        "0",
        [],
        [0],
        {},
        None,
    ]

    for v in tests:
        print(f"value={v!r:<8} type={type(v).__name__:<8} truthy={is_truthy(v)}")


if __name__ == "__main__":
    main()
